### PR TITLE
update matrix representation documentation of gates.

### PR DESCRIPTION
**Pull request by ewinston**
### Summary

Updates the matrix representation of standard gates described in their documentation to agree with their to_matrix methods if they exist.

### Details and comments
The documentation for the gates corresponds to the matrix representation of the gate with closed controls if the gate is a controlled gate. Some `to_matrix` methods are left commented out until global phase is handled in a subsequent PR.

### CODE REVIEW

File: qiskit/circuit/library/standard_gates/t.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ class TdgGate(Gate):
 
         Tdg = \begin{pmatrix}
                 1 & 0 \\
-                0 & 1-i
+                0 & e^{i\pi/4}
```
Comment (ajavadia) [inline qiskit/circuit/library/standard_gates/t.py:92]: I think a similar normalization is needed for the TGate docstring

File: qiskit/circuit/library/standard_gates/z.py [hunk 2]
```diff
@@ -176,12 +176,18 @@ def inverse(self):
...
 
-    # def to_matrix(self):
-    #     """Return a numpy.array for the CZ gate."""
-    #     return numpy.array([[1, 0, 0, 0],
-    #                         [0, 1, 0, 0],
-    #                         [0, 0, 1, 0],
-    #                         [0, 0, 0, -1]], dtype=complex)
+    def to_matrix(self):
+        """Return a numpy.array for the CZ gate."""
+        if self.ctrl_state:
```
Comment (ajavadia) [inline qiskit/circuit/library/standard_gates/z.py:181]: this should be the case for all controlled gates, not just CZ?

Comment (ewinston) [reply qiskit/circuit/library/standard_gates/z.py:181]: Yes

File: qiskit/circuit/library/standard_gates/t.py (hunk 1 above)
Comment (ewinston) [reply qiskit/circuit/library/standard_gates/t.py:92]: fixed

File: qiskit/circuit/library/standard_gates/z.py (hunk 2 above)
Comment (ewinston) [reply qiskit/circuit/library/standard_gates/z.py:181]: I added a test to verify this.

**Pull request merged**